airmozilla/main/forms.py

```diff
@@ -47,20 +47,23 @@ class Meta:
     def __init__(self, *args, **kwargs):
         self.event = kwargs.pop('event', None)
         super(EventEditForm, self).__init__(*args, **kwargs)
-        self.fields['placeholder_img'].required = False
-        self.fields['placeholder_img'].label = (
-            'Upload a picture from your computer'
-        )
-        self.fields['channels'].help_text = ""
-        self.fields['recruitmentmessage'].label = 'Recruitment message'
-        self.fields['recruitmentmessage'].required = False
-        self.fields['recruitmentmessage'].queryset = (
-            RecruitmentMessage.objects.filter(active=True)
-        )
-        self.fields['picture'].widget = GallerySelect(event=self.event)
-        self.fields['picture'].label = (
-            'Select an existing picture from the gallery'
-        )
+        if self.event.is_upcoming() or self.event.is_live():
+            self.fields['placeholder_img'].required = False
+            self.fields['placeholder_img'].label = (
+                'Upload a picture from your computer'
+            )
+            self.fields['channels'].help_text = ""
+            self.fields['recruitmentmessage'].label = 'Recruitment message'
+            self.fields['recruitmentmessage'].required = False
+            self.fields['recruitmentmessage'].queryset = (
+                RecruitmentMessage.objects.filter(active=True)
+            )
+            self.fields['picture'].widget = GallerySelect(event=self.event)
+            self.fields['picture'].label = (
+                'Select an existing picture from the gallery'
+            )
+        else:
+            del self.fields['call_info']
```

Comment:
So the only thing that should be affected by this new piece of business logic is the `call_info`. 
So leave the other lines as they were and just do:
```python
if not (self.event.is_upcoming() or self.event.is_live()):
    del self.fields['call_info']
```
```